### PR TITLE
fix: fetch sub-manifests from keppel

### PR DESCRIPTION
**Pull request by drochow**
## Description

Keppel does not scan in all cases the top layer of an image. The primary cause of that is multi-arch images. In such cases we need to list all manifests of the manifest and check for vulnerabilities on those.

In perspective, we need to decide if that is as well stored in different versions that are related to each other instead.

## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [x] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

#247 

## Added tests?

- [ ] 👍 yes
- [] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help
- [x] Separate ticket for tests # (issue/pr)

## Added to documentation?

- [ ] 📜 README.md
- [ ] 🤝 Documentation pages updated
- [x] 🙅 no documentation needed
- [ ] (if applicable) generated OpenAPI docs for CRD changes

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: scanner/keppel/main.go [hunk 1]
```diff
@@ -129,11 +129,34 @@ func HandleRepository(fqdn string, account models.Account, repository models.Rep
...
+				"repository": repository.Name,
+			}).Info("Manifest has no Vulnerabilities")
+			continue
+		}
 		HandleManifest(account, repository, manifest, component, keppelScanner, keppelProcessor)
 	}
 }
 
 func HandleManifest(account models.Account, repository models.Repository, manifest models.Manifest, component *client.Component, keppelScanner *scanner.Scanner, keppelProcessor *processor.Processor) {
+	childManifests, err := keppelScanner.ListManifestsOfManifest(account.Name, repository.Name, manifest.Digest)
```
Comment (dorneanu) [inline scanner/keppel/main.go:151]: I don't think "ListManifestsOfManifests" is a proper name for a method. Why don't you call it "ListChildManifests" whereas the 1st parameter is a manifest (followed by account.Name, repository.Name etc.)?

File: scanner/keppel/scanner/scanner.go [hunk 2]
```diff
@@ -156,6 +156,28 @@ func (s *Scanner) ListManifests(account string, repository string) ([]models.Man
 	return manifestResponse.Manifests, nil
 }
 
+func (s *Scanner) ListManifestsOfManifest(account string, repository string, manifest string) ([]models.Manifest, error) {
```
Comment (dorneanu) [inline scanner/keppel/scanner/scanner.go:159]: As pointed out a while ago, we need more comments. Without having the full context why this PR was created at all, I might need additional information (in the code) which explains why this method (in this case ListManifestsOfManifest) is needed at all.

File: scanner/keppel/main.go (hunk 1 above)
Comment (MR2011) [reply scanner/keppel/main.go:151]: I agree. This is about processing `Multi Arch` Images in Keppel. HandleMultiArchImage or something similar would be clearer IMO

File: scanner/keppel/main.go [hunk 3]
```diff
@@ -129,11 +129,34 @@ func HandleRepository(fqdn string, account models.Account, repository models.Rep
...
+
+	if err != nil {
+		log.WithFields(log.Fields{
+			"account:":   account.Name,
+			"repository": repository.Name,
+		}).WithError(err).Error("Error during ListManifestsOfManifest")
+	}
+
 	componentVersion, err := keppelProcessor.ProcessManifest(manifest, component.Id)
 	if err != nil {
```
Comment (MR2011) [inline scanner/keppel/main.go:161]: why not `if err != nil || componentVersion == nil` ?

File: scanner/keppel/main.go (hunk 1 above)
Comment (drochow) [reply scanner/keppel/main.go:151]: Well mutli arch are just the most prominent. I'd prefer "ListChildManifsts" then

Comment (drochow) [reply scanner/keppel/main.go]: done

File: scanner/keppel/scanner/scanner.go (hunk 2 above)
Comment (drochow) [reply scanner/keppel/scanner/scanner.go]: done

File: scanner/keppel/main.go (hunk 3 above)
Comment (drochow) [reply scanner/keppel/main.go:161]: done

**Review by dorneanu — APPROVED** [review]

**Pull request merged**